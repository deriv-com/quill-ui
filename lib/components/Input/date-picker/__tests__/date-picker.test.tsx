import React from "react";
import { fireEvent, render, waitFor } from "@testing-library/react";
import DatePickerDropdown from "..";
import userEvent from "@testing-library/user-event";

describe("DatePickerDropdown", () => {
    test("renders correctly", () => {
        const { container } = render(
            <DatePickerDropdown
                label="Date"
                onSelectDate={(value: Date) => {
                    return value;
                }}
            />,
        );
        expect(container.firstChild).toMatchSnapshot();
    });

    test("renders correctly when open", () => {
        const { container, getByTestId } = render(
            <DatePickerDropdown
                label="Date"
                onSelectDate={(value: Date) => {
                    return value;
                }}
                placeholder="dd/mm/yyyy"
            />,
        );

        userEvent.click(getByTestId("input-container"));
        expect(container).toMatchSnapshot();
    });

    test("renders correct label", () => {
        const { getByText } = render(
            <DatePickerDropdown
                label="Date"
                onSelectDate={(value: Date) => {
                    return value;
                }}
            />,
        );

        const label = getByText("Date");
        expect(label).toBeInTheDocument();
    });

    test("renders correct placeholder", () => {
        const { getByPlaceholderText } = render(
            <DatePickerDropdown
                onSelectDate={(value: Date) => {
                    return value;
                }}
                placeholder="dd/mm/yyyy"
            />,
        );

        const placeholder = getByPlaceholderText("dd/mm/yyyy");
        expect(placeholder).toBeInTheDocument();
    });

    test("renders correct placeholder and label", () => {
        const { getByText, getByPlaceholderText } = render(
            <DatePickerDropdown
                onSelectDate={(value: Date) => {
                    return value;
                }}
                label="Date"
                placeholder="dd/mm/yyyy"
            />,
        );

        const placeholder = getByPlaceholderText("dd/mm/yyyy");
        expect(placeholder).toBeInTheDocument();

        const label = getByText("Date");
        expect(label).toBeInTheDocument();
    });
});
