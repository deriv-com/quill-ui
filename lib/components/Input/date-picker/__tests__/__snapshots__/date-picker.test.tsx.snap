// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePickerDropdown renders correctly 1`] = `
<div
  class="datepicker__wrapper"
>
  <div
    data-testid="input-container"
  >
    <div
      class="quill-input__container"
    >
      <div
        class="quill-input__wrapper quill-input__wrapper__variant--outline quill-input__wrapper__variant--outline--neutral quill-input__wrapper__size--lg datepicker__input"
      >
        <div
          class="quill-input-icons__wrapper"
        >
          <span
            class="icon_wrapper"
          >
            <svg
              fill="var(--semantic-color-monochrome-textIcon-normal-high)"
              height="24"
              role="img"
              viewBox="0 0 32 32"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M11.625 6.5c.313 0 .625.313.625.625V9h7.5V7.125c0-.312.273-.625.625-.625.313 0 .625.313.625.625V9h1.25c1.367 0 2.5 1.133 2.5 2.5V24c0 1.406-1.133 2.5-2.5 2.5H9.75a2.47 2.47 0 0 1-2.5-2.5V11.5c0-1.367 1.094-2.5 2.5-2.5H11V7.125c0-.312.273-.625.625-.625M23.5 14h-15v10c0 .703.547 1.25 1.25 1.25h12.5c.664 0 1.25-.547 1.25-1.25zm-1.25-3.75H9.75c-.703 0-1.25.586-1.25 1.25v1.25h15V11.5c0-.664-.586-1.25-1.25-1.25"
              />
            </svg>
          </span>
          <div
            class="quill-input-label__wrapper"
          >
            <input
              class="input peer input__align--left input__size--lg"
              id="quill-input-1"
              inputmode="numeric"
              placeholder=" "
              readonly=""
              type="text"
              value=""
            />
            <label
              class="label label__status--neutral label__hasIcon"
              for="quill-input-1"
            >
              Date
            </label>
          </div>
          <svg
            class="quill-input__rotate quill-input__icon"
            data-state="close"
            height="24"
            role="img"
            viewBox="0 0 14 22"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="m6.535 14.715-5.25-5.25c-.273-.246-.273-.656 0-.93.246-.246.656-.246.93 0L7 13.348l4.785-4.786c.246-.273.656-.273.93 0a.644.644 0 0 1 0 .903l-5.277 5.25a.6.6 0 0 1-.903 0"
              />
            </g>
            <defs>
              <clippath
                id="089cb08a4a23034cce0572ced2f706b6__a"
              >
                <path
                  d="M0 0h14v22H0z"
                />
              </clippath>
            </defs>
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    class="datepicker-dropdown__menu datepicker-dropdown__menu--close datepicker-dropdown__menu--left"
  />
</div>
`;
